lib/log/global.go

```
@@ -0,0 +1,25 @@
+package log
+
+import (
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+var (
+	globalLogger     *zap.Logger
+	globalLoggerInit sync.Once
+)
+
+func getGlobal() *zapAdapter {
+	if globalLogger == nil {
+		// global logger is uninitialized - instantiate it as a dev mode logger
+		InitForTesting("info")
```

Thinking out loud, we could raise an error here to prevent dev mode logger running outside dev by: 

- checking if the binary is `go test` 
  - that's maybe a bit too crude 🤔 
- panic if the logger format is `json`, because we'll never use that format during the logs and rarely while hacking on something in local.

---

Planning on panic'ing here after some thought: https://github.com/sourcegraph/sourcegraph/pull/33956#discussion_r853232277